encoder/vaapiencoder_h264.cpp

```diff
@@ -220,10 +253,93 @@ bit_writer_write_trailing_bits(BitWriter *bitwriter)
+    bit_writer_put_ue(&scalabilityInfoWriter,
+                      temporalLayerNum - 1); // num_layers_minus1
+
+    for (i = 0; i < temporalLayerNum; i++) {
+        bit_writer_put_ue(&scalabilityInfoWriter, i); // layer_id[i]
+        scalabilityInfoWriter.writeBits(0, 6); // priority_id[i]
+        scalabilityInfoWriter.writeBits(0, 1); // discardable_flag[i]
+        scalabilityInfoWriter.writeBits(0, 3); // dependency_id[i]
+        scalabilityInfoWriter.writeBits(0, 4); // quality_id[i]
+        scalabilityInfoWriter.writeBits(i, 3); // temporal_id[i]
+        scalabilityInfoWriter.writeBits(0, 1); // sub_pic_layer_flag[i]
+        scalabilityInfoWriter.writeBits(0, 1); // sub_region_layer_flag[i]
+        scalabilityInfoWriter.writeBits(
+            0, 1); // iroi_division_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(
+            0, 1); // profile_level_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(0, 1); // bitrate_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(1, 1); // frm_rate_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(1, 1); // frm_size_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(
+            0, 1); // layer_dependency_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(
+            0, 1); // parameter_sets_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(
+            0, 1); // bitstream_restriction_info_present_flag[i]
+        scalabilityInfoWriter.writeBits(0, 1); // exact_interlayer_pred_flag[i]
+        scalabilityInfoWriter.writeBits(0, 1); // layer_conversion_flag[i]
+        scalabilityInfoWriter.writeBits(0, 1); // layer_output_flag[i]
+
+        scalabilityInfoWriter.writeBits(0, 2); // constant_frm_bitrate_idc[i]
+        scalabilityInfoWriter.writeBits((int)floor(LayerFps[i] * 256 + 0.5),
+                                        16); // avg_frm_rate
+
+        bit_writer_put_ue(&scalabilityInfoWriter,
+                          seq->picture_width_in_mbs
+                          - 1); // frm_width_in_mbs_minus1
+        bit_writer_put_ue(&scalabilityInfoWriter,
+                          seq->picture_height_in_mbs
+                          - 1); // frm_height_in_mbs_minus1
+
+        bit_writer_put_ue(&scalabilityInfoWriter,
+                          0); // layer_dependency_info_src_layer_id_delta[i]
+        bit_writer_put_ue(&scalabilityInfoWriter,
+                          0); // parameter_sets_info_src_layer_id_delta[i]
+    }
+
+    /* rbsp_trailing_bits */
+    bit_writer_write_trailing_bits(&scalabilityInfoWriter);
+
+    uint32_t scalabilityInfoBytes = scalabilityInfoWriter.getCodedBitsCount()
+                                    / 8;
+    uint8_t* scalabilityInfoData = scalabilityInfoWriter.getBitWriterData();
+    ASSERT(scalabilityInfoBytes && scalabilityInfoData);
+
+    bit_writer_write_nal_header(bitwriter, VAAPI_ENCODER_H264_NAL_REF_IDC_NONE,
```

Comment:
We can keep the position of scalabilityInfoBytes, rewrite it after bit_writer_write_trailing_bits.